src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -680,17 +681,13 @@ void DoorLockServer::getCredentialStatusCommandHandler(chip::app::CommandHandler
     }
 
     uint16_t maxNumberOfCredentials = 0;
+    EmberAfPluginDoorLockCredentialInfo credentialInfo;
     if (!credentialIndexValid(commandPath.mEndpointId, credentialType, credentialIndex, maxNumberOfCredentials))
     {
-        emberAfDoorLockClusterPrintln("[GetCredentialStatus] Credential index is out of range "
-                                      "[endpointId=%d,credentialType=%u,credentialIndex=%d,maxNumberOfCredentials=%d]",
-                                      commandPath.mEndpointId, to_underlying(credentialType), credentialIndex,
-                                      maxNumberOfCredentials);
-        emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_INVALID_COMMAND);
+        sendGetCredentialResponse(commandObj, commandPath, credentialType, credentialIndex, 0, credentialInfo, false);
```

Comment:
Should credentialInfo be initialized by this point?

Reply:
Currently credential info is initialized by emberAfPluginDoorLockGetCredential with a valid index. 

Now that we want to return a failure response instead of INVALID_COMMAND, maybe we should look at initializing credentialInfo with some default values so it can be safely passed from function to function

Reply:
Maybe the easier change would be to change the `credentialInfo`'s parameter type in `sendGetCredentialResponse` to `Optional<EmberAfPluginDoorLockCredentialInfo>`? Or maybe to just `EmberAfPluginDoorLockCredentialInfo *` and pass a `nullptr` in case we don't have any info?